src/ansi-c/cprover_library.cpp

```diff
@@ -45,7 +45,9 @@ std::string get_cprover_library_text(
   const struct cprover_library_entryt cprover_library[],
   const std::string &prologue)
 {
-  std::ostringstream library_text(prologue);
+  // the default mode is ios_base::out which means subsequent write to the
+  // stream will overwrite the original content
+  std::ostringstream library_text(prologue, std::ios_base::ate);
```

Comment:
```
#include <iostream>
#include <sstream>
int main(void) {
  std::ostringstream send_help{"SOS", std::ios::ate};
  send_help << "WAT";
  std::cout << send_help.str() << '\n';
}
```

Reply:
WAT is the point of this constructor... 
A potentially more obvious fix might be 
```
std::ostringstream library_text;
library_text << prologue
```

Reply:
@thk123 In a slightly less humorous note, I _can_ see uses for this (for example, if you need to guarantee some minimum width for the output so you add in a bunch of filler characters to override first), the “wtf” here is that this is the default, which I think few sane people would expect.

Reply:
```
std::ostringstream library_text(prologue);
library_text.seekp(0, library_text.end);
```
would also work.

Reply:
I'd personally lean either towards the solution @thk123 suggested, or to the original solution @xbauch had in the PR. I think the alternative using `seekp` just looks even more odd :-) Though I appreciate the oddness comes from the C++ spec here.